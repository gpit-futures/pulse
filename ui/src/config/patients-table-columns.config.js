import React from 'react';

import { getDDMMMYYYY } from '../utils/time-helpers.utils';

const CalendarIcon = () => <i className="fa fa-calendar" />;

export const patientsColumnsConfig = [
  { key: 'name', title: 'Name', width: 150 },
  { key: 'address', title: 'Address', width: 300 },
  { key: 'dateOfBirth', title: 'Born', transformer: getDDMMMYYYY, width: 105 },
  { key: 'promsCount', title: 'PROM ', icon: <span>#</span>, width: 100 },
  { key: 'promsDate', title: 'PROM', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'gender', title: 'Gender', width: 90 },
  { key: 'id', title: 'NHS No.', width: 115 },
  { key: 'ordersDate', title: 'Orders', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'ordersCount', title: 'Orders ', icon: <span>#</span>, width: 100 },
  { key: 'resultsDate', title: 'Results', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'resultsCount', title: 'Results ', icon: <span>#</span>, width: 100 },
  { key: 'vitalsDate', title: 'Vitals', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'vitalsCount', title: 'Vitals ', icon: <span>#</span>, width: 100 },
  { key: 'diagnosesDate', title: 'Diagnosis', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 130 },
  { key: 'diagnosesCount', title: 'Diagnosis ', icon: <span>#</span>, width: 120 },
  { key: 'viewPatientNavigation', title: '', icon: <span />, width: 90 },
];

export const defaultColumnsSelected = {
  name: true,
  gender: true,
  dateOfBirth: true,
  address: true,
  id: true,
  ordersDate: true,
  resultsDate: true,
  vitalsDate: false,
  diagnosesDate: false,
  ordersCount: true,
  resultsCount: true,
  vitalsCount: false,
  diagnosesCount: false,
  viewPatientNavigation: true,
  promsCount: false,
  promsDate: false,
};
