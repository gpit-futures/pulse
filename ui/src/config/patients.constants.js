import { getAgeYears } from '../utils/time-helpers.utils';

export const AGE_RANGE_PREFIX = 'ageRange';
export const DEPARTMENT_PREFIX = 'department';

export const patientsDepartmentPredicate = departmentName => patient => patient.department === departmentName;

export const patientsAgeRangesPredicate = (from, to) => patient => from <= getAgeYears(patient.dateOfBirth) && getAgeYears(patient.dateOfBirth) <= to;

export const patientsDepartments = [
  { name: 'Community Care', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Community Care') },
  { name: 'Hospital', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Hospital') },
  { name: 'Mental Health', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Mental Health') },
  { name: 'Neighbourhood', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Neighbourhood') },
  { name: 'Primary Care', prefix: DEPARTMENT_PREFIX, prefixLabel: 'Setting', predicate: patientsDepartmentPredicate('Primary Care') },
];

export const patientsAgeRanges = [
  { name: '19-30', from: 19, to: 30, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(19, 30) },
  { name: '31-60', from: 31, to: 60, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(31, 60) },
  { name: '61-80', from: 61, to: 80, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(61, 80) },
  { name: '>80', from: 81, to: 999, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(81, 999) },
];

const ageRangeForClinicalSearch = [{ name: '11-18', from: 11, to: 18, prefix: AGE_RANGE_PREFIX, prefixLabel: 'Age range', predicate: patientsAgeRangesPredicate(11, 18) }];

export const patientAgeRangesForClinicalSearch = ageRangeForClinicalSearch.concat(patientsAgeRanges);