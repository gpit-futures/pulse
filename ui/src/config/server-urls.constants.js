export const apiUrls = {
  INITIALISE: '/api/initialise',
  LOGOUT: '/api/logout',
};

export const usersUrls = {
  PATIENTS_URL: '/api/patients',
  USER_ACCOUNT_URL: '/api/user',
  BASIC_PATIENT_SEARCH: '/api/search/patient/table',
  ADVANCED_PATIENT_SEARCH: '/api/patients/advancedSearch',
  PROFILE_APP_PREFERENCES: '/api/application',
  LIST_ORDERS: '/api/terminology/list/order',
  PICTURES: '/api/pictures',
  CLINICAL_QUERY_SEARCH: '/api/patients/querySearch',
  FEEDS: '/api/feeds',
};
