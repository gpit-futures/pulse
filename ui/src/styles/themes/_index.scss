@import 'gray';
@import 'purple';
@import 'leedsPHR';
