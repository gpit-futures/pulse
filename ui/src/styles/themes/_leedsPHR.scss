$dodger-blue: #3596f4;
$lucky-point: #18276d;
$white: #ffffff;

$base-color: $dodger-blue;
$auxiliary-color: $lucky-point;

$table-color: lighten($base-color, 40%);;
$table-text: #5c5c5c;
$colored-table: (
  bg: #ffffff,
  color: $table-text,
  border-color: #d4d2d2,
  border-width: 1px,

  header-bg: #e0e0e0,
  header-color: #333,
  header-highlighted-bg: darken(#e0e0e0, 10%),
  header-highlighted-color: #333,

  row-bg: #ffffff,
  row-odd-bg: #ffffff,
  row-hover-bg: $auxiliary-color,
  row-odd-hover-bg: $auxiliary-color,
  row-color: $table-text,
  row-odd-color: $table-text,
  row-hover-color: #fff,
  row-odd-hover-color: #fff,

  row-sorted-bg: lighten(#e0e0e0, 7%),
  row-sorted-odd-bg: lighten(#e0e0e0, 7%),
  row-sorted-hover-bg: $auxiliary-color,
  row-sorted-odd-hover-bg: $auxiliary-color,
  row-sorted-color: $table-text,
  row-sorted-odd-color: $table-text,
  row-sorted-hover-color: #fff,
  row-sorted-odd-hover-color: #fff,

  row-highlighted-bg: #b4b4b4,
  row-highlighted-color: #ffffff,
);

$text-color: #303030;
$text-on-base-color: #ffffff;

$colored-buttons: (
    default: #ffffff,
    primary: #337AB7,
    success: #017A41,
    info: #31b0d5,
    warning: #F0AD4E,
    danger: #DA534F,
    theme: $auxiliary-color
);


/**
  * COLORS for base configuration
  */
$body-bg: #eee;

/**
  * COLORS for header
  */

$header-bg: $white;
$header-btn: (
    bg: $dodger-blue,
    color: $text-on-base-color,
    bg-hover: $lucky-point,
    color-hover: $text-on-base-color,
    bg-active: $lucky-point,
    color-active: $text-on-base-color
);
$header-toolbar: (
    bg: $lucky-point,
    color: $text-on-base-color,
);

/**
  * COLORS for search
  */
  $search-btn: (
      bg: $base-color,
      color: $text-on-base-color,
      bg-hover: $auxiliary-color,
      color-hover: $text-on-base-color
  );
  $search-btn-toggle: (
      bg: $dodger-blue,
      color: $text-on-base-color,
      bg-hover: $auxiliary-color,
      color-hover: $text-on-base-color,
      border: $base-color,
      border-hover: $auxiliary-color
  );

/**
  * COLORS for footer
  */
  $footer: (
    bg: $white,
    color: #5c5c5c
  );

/**
  * COLORS for dashboard
  */
  $board: (
    border: #c9c8c8,
    radius: 4px,
    header-bg: $base-color,
    header-color: $text-on-base-color,
    preview-ratio: 60%,
    item-bg: $white,
    item-odd-bg: $white,
    item-color: $text-color,
    item-bg-hover: $auxiliary-color,
    item-color-hover: $white,
  );

/**
  * COLORS for panels
  */
  $panel-border-color: #d4d2d2;
  $panel-heading-bg: $base-color;
  $panel-heading-color: $text-on-base-color;
  $panel-control-bg: transparent;
  $panels: (
    panel-default: (
      bg: #fff,
      heading-bg: $base-color,
      heading-color: $text-on-base-color,
      border-color: $base-color,
      border-width: 2px,
      border-radius: 3px,
      control-bg: $panel-control-bg
    ),
    panel-primary: (
      bg: transparent,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: transparent,
      border-width: 0,
      border-radius: 3px,
      control-bg: transparent
    ),
    panel-secondary: (
      bg: #fff,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: $panel-border-color,
      border-width: 1px,
      border-radius: 3px,
      control-bg: $panel-control-bg
    ),
    panel-subaccordion: (
      bg: #fff,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: transparent,
      border-width: 0,
      border-radius: 3px,
      control-bg: $panel-control-bg
    ),
    panel-small: (
      bg: #fff,
      heading-bg: $panel-heading-bg,
      heading-color: $panel-heading-color,
      border-color: $panel-border-color,
      border-width: 1px,
      border-radius: 3px,
      control-bg: $panel-control-bg
    )
  );

/**
  * COLORS for forms
  */
  $forms: (
    bg: #fff,
    color: #555555,
    disabled-bg: #ebebeb,
    disabled-color: #5c5c5c,
    readonly-bg: #fff,
    readonly-color: #555,

    border-color: #c5c4c4,
    border-error-color: #a94442,
    border-success-color: #2b542c,
    border-width: 1px,
    border-radius: 4px,
    addon-color: #222,
    label-color: #333,
    label-control-color: #333,
    label-required-color: #a32d44,
    control-static-color: #333,
  );

body.themeLeedsPHR {
  @import 'theme-source/index';
  @import 'theme-leedsPHR/index';
}