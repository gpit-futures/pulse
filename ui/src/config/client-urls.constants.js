import { themeConfigs } from '../themes.config';

export const clientUrls = {
  ROOT: '/',
  CHARTS: '/charts',
  PATIENTS: '/patients',
  PATIENTS_FULL_DETAILS: '/patients-full-details',
  SEARCH_REPORT: '/search-report',
  USER_PROFILE: '/profile',
  PATIENTS_SUMMARY: 'patients-summary',
  RESULTS: 'results',
  DIAGNOSES: 'diagnoses',
  ALLERGIES: 'allergies',
  CONTACTS: 'contacts',
  CLINICAL_NOTES: 'clinicalNotes',
  PERSONAL_NOTES: 'personalNotes',
  VACCINATIONS: 'vaccinations',
  MEDICATIONS: 'medications',
  GENERIC_PLUGIN: 'genericPlugin',
  PROCEDURES: 'procedures',
  EVENTS: 'events',
  TEST_RESULTS: 'results',
  REFERRALS: 'referrals',
  ORDERS: 'orders',
  MDTS: 'mdt',
  DRAWINGS: 'drawings',
  VITALS: 'vitals',
  DOCUMENTS: 'documents',
  IMAGES: 'images',
  TRANSFERS_OF_CARE: 'transfer-of-care',
  TOP_THREE_THINGS: 'topThreeThings',
  CLINICAL_STATEMENTS: 'clinicalStatements',
  PROMS: 'proms',
  FEEDS: 'feeds',
};

const pluginsPages = {
  'allergies': {
    breadcrumbs: [{
      title: 'Allergies',
      state: '/allergies',
    }],
  },
  'diagnoses': {
    breadcrumbs: [{
      title: 'Problems / Diagnoses',
      state: '/diagnoses',
    }],
  },
  'clinicalNotes': {
    breadcrumbs: [{
      title: 'Clinical Notes',
      state: '/clinicalNotes',
    }],
  },
  'personalNotes': {
    breadcrumbs: [{
      title: 'Personal Notes',
      state: '/personalNotes',
    }],
  },
  'contacts': {
    breadcrumbs: [{
      title: 'Contacts',
      state: '/contacts',
    }],
  },
  'vaccinations': {
    breadcrumbs: [{
      title: 'Vaccinations',
      state: '/vaccinations',
    }],
  },
  'genericPlugin': {
    breadcrumbs: [{
      title: 'Generic Plugin',
      state: '/genericPlugin',
    }],
  },
  'medications': {
    breadcrumbs: [{
      title: 'Medications',
      state: '/medications',
    }],
  },
  'procedures': {
    breadcrumbs: [{
      title: 'Procedures',
      state: '/procedures',
    }],
  },
  'events': {
    breadcrumbs: [{
      title: 'Events',
      state: '/events',
    }],
  },
  'results': {
    breadcrumbs: [{
      title: 'Test Results',
      state: '/results',
    }],
  },
  'referrals': {
    breadcrumbs: [{
      title: 'Referrals',
      state: '/referrals',
    }],
  },
  'orders': {
    breadcrumbs: [{
      title: 'Orders',
      state: '/orders',
    }],
  },
  'mdt': {
    breadcrumbs: [{
      title: 'MDT',
      state: '/mdt',
    }],
  },
  'drawings': {
    breadcrumbs: [{
      title: 'Drawings',
      state: '/drawings',
    }],
  },
  'vitals': {
    breadcrumbs: [{
      title: 'Vitals - NEWS',
      state: '/vitals',
    }],
  },
  'documents': {
    breadcrumbs: [{
      title: 'Documents',
      state: '/documents',
    }],
  },
  'images': {
    breadcrumbs: [{
      title: 'Images',
      state: '/images',
    }],
  },
  'transfer-of-care': {
    breadcrumbs: [{
      title: 'Transfers Of Care',
      state: '/transfers-of-care',
    }],
  },
  'topThreeThings': {
    breadcrumbs: [{
      title: 'Top 3 Things',
      state: '/topThreeThings',
    }],
  },
  'clinicalStatements': {
    breadcrumbs: [{
      title: 'Clinical Statements',
      state: '/clinicalStatements',
    }],
  },
  'proms': {
    breadcrumbs: [{
      title: 'PROMs',
      state: '/proms',
    }],
  },
};


const addPluginsPagesToLists = (list, breadcrumbsBefore) => {
  for (const nameOfPage in pluginsPages) {
    const breadcrumbs = breadcrumbsBefore.concat(pluginsPages[nameOfPage].breadcrumbs);
    list[nameOfPage] = { breadcrumbs };
  }
};

export const mainPagesTitles = {
  '/': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/',
    }],
    headerTitle: 'System Dashboard',
  },
  'charts': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/charts',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Patient Listings',
      state: '/patients',
    }],
    headerTitle: 'Patients Lists',
  },
  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'search-report': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Search Report',
      state: '/search-report',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: 'Patient Listings',
      state: '/patients',
    }, {
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
  'patients-full-details': {
    breadcrumbs: [],
    headerTitle: 'Search results',
  },
};
addPluginsPagesToLists(mainPagesTitles, [{
  title: 'Patient Listings',
  state: '/patients',
}, {
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);

export const mainPagesTitlesForPatients = {
  '/': {
    breadcrumbs: [{
      title: 'Patient Summary',
      state: '/patients-summary',
    }],
    headerTitle: 'Patient Summary',
  },

  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
};
addPluginsPagesToLists(mainPagesTitlesForPatients, [{
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);
