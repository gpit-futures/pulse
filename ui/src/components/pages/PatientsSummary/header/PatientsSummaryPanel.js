import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';
import { lifecycle } from 'recompose';
import { Row } from 'react-bootstrap';
import { Col } from 'react-bootstrap';

import PTCustomInput from './PTCustomInput';
import { unmountOnBlur } from '../../../../utils/HOCs/unmount-on-blur.utils'
import { patientsSummaryConfig } from '../patients-summary.config';
import { themeConfigs } from '../../../../themes.config';
import { dashboardBeing } from '../../../../plugins.config';
import { getNameFromUrl } from '../../../../utils/rss-helpers';

@lifecycle(unmountOnBlur)
export default class PatientsSummaryPanel extends PureComponent {
    static propTypes = {
      onCategorySelected: PropTypes.func.isRequired,
      onViewOfBoardsSelected: PropTypes.func.isRequired,
      selectedCategory: PropTypes.objectOf(PropTypes.bool).isRequired,
      selectedViewOfBoards: PropTypes.objectOf(PropTypes.bool).isRequired,
    };

    state = {
      selected: this.props.selectedCategory,
      selectedViewOptions: this.props.selectedViewOfBoards,
    };

    componentDidUpdate(prevProps, prevState) {
      if (!_.isEqual(prevState.selected, this.state.selected)) this.props.onCategorySelected(this.state.selected);
      if (!_.isEqual(prevState.selectedViewOptions, this.state.selectedViewOptions)) this.props.onViewOfBoardsSelected(this.state.selectedViewOptions);
    }

    toggleCheckbox = key => this.setState((prevState) => {
      const newValue = !_.get(['selected', key])(prevState);
      return _.set(['selected', key], newValue)(prevState);
    });

    toggleRadio = key => this.setState((prevState) => {
      const selectedViewOptions = Object.assign({}, prevState.selectedViewOptions);
      selectedViewOptions.full = false;
      selectedViewOptions.preview = false;
      selectedViewOptions.list = false;
      selectedViewOptions[key] = true;
      return { 'selectedViewOptions': selectedViewOptions };
    });

    render() {
      const { selected, selectedViewOptions } = this.state;
      const { patientsSummaryHasPreviewSettings, feeds } = this.props;

      return (
        <div className="dropdown-menu dropdown-menu-panel dropdown-menu-summary">
          <div className="form-group-wrapper">
            <div className="heading">SHOW</div>
            <div className="form-group">
              <Row>
                {patientsSummaryConfig.map((item) => {
                  return dashboardBeing[item.key] !== false ?
                    <Col xs={6} sm={4} key={item.nameCheckboxes}>
                      <PTCustomInput
                        type="checkbox"
                        title={item.titleCheckboxes}
                        id={item.nameCheckboxes}
                        name={item.nameCheckboxes}
                        isChecked={selected[item.key]}
                        onChange={this.toggleCheckbox}
                      />
                    </Col> : null
                })}
              </Row>
            </div>
            {themeConfigs.isLeedsPHRTheme ?
              <div>
                <div className="heading">FEEDS</div>
                <div className="form-group">
                  <Row>
                    {feeds.map((item) => {
                      const nameItem = getNameFromUrl(item.landingPageUrl);
                      return (
                        <Col xs={6} sm={4} key={nameItem}>
                          <PTCustomInput
                            type="checkbox"
                            title={item.name}
                            id={nameItem}
                            name={nameItem}
                            isChecked={selected[nameItem] || false}
                            onChange={this.toggleCheckbox}
                          />
                        </Col>)
                    })}
                  </Row>
                </div>
              </div> : null }
            {(themeConfigs.patientsSummaryHasPreviewSettings || patientsSummaryHasPreviewSettings) ?
              <div>
                <div className="heading">VIEW OF BOARDS</div>
                <div className="form-group">
                  <Row>
                    <Col xs={12}>
                      <PTCustomInput type="radio" title="Full View" id="full" name="view-of-preview" value="full" isChecked={selectedViewOptions.full} onChange={this.toggleRadio} />
                    </Col>
                  </Row>
                  <Row>
                    <Col xs={12} sm={6}>
                      <PTCustomInput type="radio" title="Only Preview" id="preview" name="view-of-preview" value="preview" isChecked={selectedViewOptions.preview} onChange={this.toggleRadio} />
                    </Col>
                    <Col xs={12} sm={6}>
                      <PTCustomInput type="radio" title="Only List" id="list" name="view-of-preview" value="list" isChecked={selectedViewOptions.list} onChange={this.toggleRadio} />
                    </Col>
                  </Row>
                </div>
              </div> : null}
          </div>
        </div>
      )
    }
}
