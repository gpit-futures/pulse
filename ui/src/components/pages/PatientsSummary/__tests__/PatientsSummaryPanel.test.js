import React from 'react';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15';

import PatientsSummaryPanel from '../header/PatientsSummaryPanel';
import { themeConfigs } from '../../../../themes.config';

const testProps = {
  onCategorySelected: () => {},
  selectedCategory: {
    problems: true,
    contacts: true,
    allergies: true,
    medications: true,
  },
  feeds: [
    {
      name: 'NYTimes.com',
      landingPageUrl: 'https://www.nytimes.com/section/health',
      rssFeedUrl: 'http://rss.nytimes.com/services/xml/rss/nyt/Health.xml',
      sourceId: 'testSourceID6',
    }, {
      name: 'BBC Health',
      landingPageUrl: 'http://www.bbc.co.uk/news/health',
      rssFeedUrl: 'http://feeds.bbci.co.uk/news/health/rss.xml?edition=uk#',
      sourceId: 'testSourceID1',
    }, {
      name: 'NHS Choices',
      landingPageUrl: 'https://www.nhs.uk/news/',
      rssFeedUrl: 'https://www.nhs.uk/NHSChoices/shared/RSSFeedGenerator/RSSFeed.aspx?site=News',
      sourceId: 'testSourceID2',
    }, {
      name: 'Public Health',
      landingPageUrl: 'https://www.gov.uk/government/organisations/public-health-england',
      rssFeedUrl: 'https://www.gov.uk/government/organisations/public-health-england.atom',
      sourceId: 'testSourceID3',
    }, {
      name: 'Leeds Live - Whats on',
      landingPageUrl: 'https://www.leeds-live.co.uk/best-in-leeds/whats-on-news/',
      rssFeedUrl: 'https://www.leeds-live.co.uk/best-in-leeds/whats-on-news/?service=rss',
      sourceId: 'testSourceID4',
    }, {
      name: 'Leeds CC Local News',
      landingPageUrl: 'https://news.leeds.gov.uk',
      rssFeedUrl: 'https://news.leeds.gov.uk/tagfeed/en/tags/Leeds-news',
      sourceId: 'testSourceID5',
    },
  ],
};

configure({ adapter: new Adapter() });

describe('Component <PatientsSummaryPanel />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <PatientsSummaryPanel
        onCategorySelected={testProps.onCategorySelected}
        selectedCategory={testProps.selectedCategory}
        onViewOfBoardsSelected={() => {}}
        feeds={testProps.feeds}
      />).dive();
    expect(component).toMatchSnapshot();

    expect(component.instance().props.onCategorySelected).toEqual(testProps.onCategorySelected);
    expect(component.instance().props.selectedCategory).toEqual(testProps.selectedCategory);

    expect(component.find('.heading')).toHaveLength(1);
    expect(component.find('.heading').text()).toEqual('SHOW');
    expect(component.find('.form-group')).toHaveLength(1);
    expect(component.find('PTCustomInput')).toHaveLength(4);

    component.instance().toggleCheckbox('dashboard-name');
    component.setState({ selected: {
      problems: true,
      contacts: false,
      allergies: true,
      medications: false,
    } });
    component.setState({ selected: {
      problems: true,
      contacts: false,
      allergies: true,
      medications: false,
    } });
    expect(component.find('PTCustomInput')).toHaveLength(4);

    component.instance().toggleRadio('test');

    component.setState({
      selectedViewOptions: {
        full: false,
        preview: true,
      },
    })

    component.setProps({
      patientsSummaryHasPreviewSettings: true,
    })
  });

  it('should renders with all props correctly with LeedsPHRTheme', () => {
    themeConfigs.isLeedsPHRTheme = true;
    const component = shallow(
      <PatientsSummaryPanel
        onCategorySelected={testProps.onCategorySelected}
        selectedCategory={testProps.selectedCategory}
        onViewOfBoardsSelected={() => {}}
        feeds={testProps.feeds}
      />).dive();

    expect(component).toMatchSnapshot();

  });
});
