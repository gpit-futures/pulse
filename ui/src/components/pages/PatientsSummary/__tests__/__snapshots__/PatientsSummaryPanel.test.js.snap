// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PatientsSummaryPanel /> should renders with all props correctly 1`] = `
<div
  className="dropdown-menu dropdown-menu-panel dropdown-menu-summary"
>
  <div
    className="form-group-wrapper"
  >
    <div
      className="heading"
    >
      SHOW
    </div>
    <div
      className="form-group"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          key="problems"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="problems"
            isChecked={true}
            name="problems"
            onChange={[Function]}
            title="Problems"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="contacts"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="contacts"
            isChecked={true}
            name="contacts"
            onChange={[Function]}
            title="Contacts"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="allergies"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="allergies"
            isChecked={true}
            name="allergies"
            onChange={[Function]}
            title="Allergies"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="medications"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="medications"
            isChecked={true}
            name="medications"
            onChange={[Function]}
            title="Medications"
            type="checkbox"
          />
        </Col>
      </Row>
    </div>
  </div>
</div>
`;

exports[`Component <PatientsSummaryPanel /> should renders with all props correctly with LeedsPHRTheme 1`] = `
<div
  className="dropdown-menu dropdown-menu-panel dropdown-menu-summary"
>
  <div
    className="form-group-wrapper"
  >
    <div
      className="heading"
    >
      SHOW
    </div>
    <div
      className="form-group"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          key="problems"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="problems"
            isChecked={true}
            name="problems"
            onChange={[Function]}
            title="Problems"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="contacts"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="contacts"
            isChecked={true}
            name="contacts"
            onChange={[Function]}
            title="Contacts"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="allergies"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="allergies"
            isChecked={true}
            name="allergies"
            onChange={[Function]}
            title="Allergies"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="medications"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="medications"
            isChecked={true}
            name="medications"
            onChange={[Function]}
            title="Medications"
            type="checkbox"
          />
        </Col>
      </Row>
    </div>
    <div>
      <div
        className="heading"
      >
        FEEDS
      </div>
      <div
        className="form-group"
      >
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            key="nytimes_com"
            sm={4}
            xs={6}
          >
            <PTCustomInput
              id="nytimes_com"
              isChecked={false}
              name="nytimes_com"
              onChange={[Function]}
              title="NYTimes.com"
              type="checkbox"
            />
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            key="bbc_co_uk"
            sm={4}
            xs={6}
          >
            <PTCustomInput
              id="bbc_co_uk"
              isChecked={false}
              name="bbc_co_uk"
              onChange={[Function]}
              title="BBC Health"
              type="checkbox"
            />
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            key="nhs_uk"
            sm={4}
            xs={6}
          >
            <PTCustomInput
              id="nhs_uk"
              isChecked={false}
              name="nhs_uk"
              onChange={[Function]}
              title="NHS Choices"
              type="checkbox"
            />
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            key="gov_uk"
            sm={4}
            xs={6}
          >
            <PTCustomInput
              id="gov_uk"
              isChecked={false}
              name="gov_uk"
              onChange={[Function]}
              title="Public Health"
              type="checkbox"
            />
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            key="leeds-live_co_uk"
            sm={4}
            xs={6}
          >
            <PTCustomInput
              id="leeds-live_co_uk"
              isChecked={false}
              name="leeds-live_co_uk"
              onChange={[Function]}
              title="Leeds Live - Whats on"
              type="checkbox"
            />
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            key="news_leeds_gov_uk"
            sm={4}
            xs={6}
          >
            <PTCustomInput
              id="news_leeds_gov_uk"
              isChecked={false}
              name="news_leeds_gov_uk"
              onChange={[Function]}
              title="Leeds CC Local News"
              type="checkbox"
            />
          </Col>
        </Row>
      </div>
    </div>
  </div>
</div>
`;
